File: java/test/src/main/java/test/Ice/proxy/CI.java
// Copyright (c) ZeroC, Inc.

package test.Ice.proxy;

import com.zeroc.Ice.Current;

import test.Ice.proxy.Test.APrx;
import test.Ice.proxy.Test.BPrx;
import test.Ice.proxy.Test.C;
import test.Ice.proxy.Test.CPrx;
import test.Ice.proxy.Test.S;

public class CI implements C {
    public APrx opA(APrx a, Current current) {
        return a;
    }

    public BPrx opB(BPrx b, Current current) {
        return b;
    }

    public CPrx opC(CPrx c, Current current) {
        return c;
    }

    public S opS(S s, Current current) {
        return s;
    }
}

File: csharp/test/Ice/proxy/CI.cs
// Copyright (c) ZeroC, Inc.

#nullable enable

namespace Ice.proxy;

public sealed class CI : Test.CDisp_, Ice.Object
{
    public CI()
    {
    }

    public override Test.APrx? opA(Test.APrx? a, Ice.Current current) => a;
    public override Test.BPrx? opB(Test.BPrx? b, Ice.Current current) => b;
    public override Test.CPrx? opC(Test.CPrx? c, Ice.Current current) => c;
    public override Test.S opS(Test.S s, Ice.Current current) => s;
}
